ambry-commons/src/main/java/com.github.ambry.commons/BlobIdBuilder.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright 2017 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+package com.github.ambry.commons;
+
+import com.github.ambry.account.Account;
+import com.github.ambry.account.Container;
+import com.github.ambry.clustermap.PartitionId;
+
+
+/**
+ * A builder class for {@link BlobId}. This class is not thread safe.
+ */
+public class BlobIdBuilder {
```

Comment:
Per our discussion offline, it is preferable to keep this builder for mainly two reasons:
1. We will have a single controller, that is the `CURRENT_VERSION`, to control the serialized version for `BlobId`s constructed by providing field arguments. We will not use constructors to control the version. So There won't be any code change if we want to change the serialization format;
2. The field can be null, and it is at the `BlobId` level to decide the values for missing fields. This will be better than the caller to know the default/legacy values for any N/A fields. Therefore, if the caller does not have the value for a missing field, it just supplies null or does not set the field, the `BlobId` will resolve it.